assets/docref/docref.js

```diff
@@ -21,7 +21,8 @@ $(function() {
       // Wait 250ms and scroll to anchor after loading
       setTimeout(function() {
         var hash = (window.location.hash || "").substr(1);
-        if(hash && hash !== "") {
+        // be a little careful about what gets interpolated into the jQuery here
+        if(hash && hash.match(/[a-zA-Z0-9_ -]+/)) {
```

Comment:
If the goal is to restrict what could be present in hash to prevent malicious eval, we should anchor the start and end of that regex to prevent code exec.  Otherwise, it will simply match on anything that has one of those chars and be easily bypassed by "baddies".

Current /[a-zA-Z0-9_ -]+/
- 'window' <= Match
- 'window.location = "evil.com"' <= Match

Proposed /^[a-zA-Z0-9_ -]+$/
- 'window'  <= Match
- 'window.location = "evil.com"' <= NoMatch